mosfit/modules/seds/blackbody.py

```diff
@@ -28,37 +31,51 @@ def process(self, **kwargs):
         self._temperature_phot = kwargs['temperaturephot']
         xc = self.X_CONST
         fc = self.FLUX_CONST
+        xcv = self.X_CONST_FREQ
+        fcv = self.FLUX_CONST_FREQ
         cc = self.C_CONST
         temperature_phot = self._temperature_phot
         zp1 = 1.0 + kwargs['redshift']
         seds = []
         evaled = False
         for li, lum in enumerate(self._luminosities):
+            radius_phot = self._radius_phot[li]
+            temperature_phot = self._temperature_phot[li]
             bi = self._band_indices[li]
             if lum == 0.0:
                 if bi >= 0:
-                    seds.append(np.zeros_like(self._sample_frequencies[bi]))
+                    seds.append(np.zeros_like(self._sample_wavelengths[bi]))
                 else:
                     seds.append([0.0])
                 continue
             if bi >= 0:
-                rest_freqs = self._sample_frequencies[bi] * zp1
-            else:
-                rest_freqs = [self._frequencies[li] * zp1]
-            radius_phot = self._radius_phot[li]
-            temperature_phot = self._temperature_phot[li]
+                rest_wavs = (self._sample_wavelengths[bi]
+                            * u.Angstrom.cgs.scale / zp1)
+                if not evaled:
+                    sed = ne.evaluate(
+                        'fc * radius_phot**2 / rest_wavs**5 / '
+                        '(exp(xc / rest_wavs / temperature_phot) - 1.0)')
+                    evaled = True
+                else:
```

Comment:
This isn't safe now, you have two different numexpr arguments for radio/non-radio, so a re-evaluate will be in error if switching between radio/non-radio observations. The numexpr should be the same for both (so I'd convert to wavelength before passing into numexpr).